alt_e2eshark/run.py

```diff
@@ -66,8 +67,17 @@ def get_tests(groups, test_filter):
 
 def main(args):
     """Sets up config and test list based on CL args, then runs the tests"""
-    # TODO: allow for no-run/mark xfails
-    # TODO: add better logging setup
+
+    # set-up cache directory. Tries to get from env. var.
+    CACHE_DIR = os.getenv('CACHE_DIR')
+    cache_dir = args.cachedir
+    if not CACHE_DIR and not cache_dir:
+        raise RuntimeError("No CACHE_DIR environment variable set, and no --cachedir arg provided.")
+    if cache_dir:
+        # if a --cachedir arg is provided, use it for the tests and update the env variable.
+        os.environ['CACHE_DIR'] = cache_dir
```

Comment:
I don't like this because typically linux commands with arguments like this overrides the env var for a single run and does not update the env variable.

To allow the user to set a default for all future runs, I would much rather read a bunch of variables called something like SHARKTEST_XXX and have users put them into a .env file.

Reply:
making --cachedir set the environment variable might cause surprises & make for weird persistence issues. I think we should let users manage their own environment variables and include info on the environment variables in the --help output